platform/scheduler/zephyr/src/mender-scheduler.c

```diff
@@ -59,6 +89,17 @@ mender_scheduler_alt_work_create(mender_scheduler_alt_work_function_t func, int3
     user_function = func;
     user_interval = interval;
 
+#ifdef CONFIG_MENDER_SCHEDULER_SEPARATE_WORK_QUEUE
+    /* Create and start work queue */
+    k_work_queue_init(&mender_scheduler_work_queue_handle);
+    k_work_queue_start(&mender_scheduler_work_queue_handle,
+                       mender_scheduler_work_queue_stack,
+                       CONFIG_MENDER_SCHEDULER_WORK_QUEUE_STACK_SIZE * 1024,
+                       CONFIG_MENDER_SCHEDULER_WORK_QUEUE_PRIORITY,
+                       NULL);
+    k_thread_name_set(k_work_queue_thread_get(&mender_scheduler_work_queue_handle), "mender_scheduler_work_queue");
```

Comment:
I would make the thread name shorter. This is printed for example in the stack overflow error messages, and from what I've seen are just short ids.

What about "mender_queue" or "mender_work"?

Reply:
I'm changing this in the follow-up to this and there I name the work queue which is what we should actually do, I 
believe.